### PR TITLE
Upgrade dependencies for bug and for security vulnerabilities.

### LINKED ISSUE #4014: [Bug] Update `etf/pir` to v1.1.2 and new functionality
This should fix the current error it is showing as it relied on an older version of `yfinance`. I also made the report look neater, cleaning up some naming. It also shouldn't be marked `grey` as the command runs independently from the `load` functionality.

image

Furthermore, this release also contains new functionality. Up to you whether it is relevant to incorporate in the OpenBB Terminal and/or OpenBB SDK. See below.

```python
import thepassiveinvestor as pi

# Collect data from a set of ETFs and compare them
etf_comparison = pi.collect_data(['VOO', 'QQQ', 'ARKG', 'VUG', 'SCHA', 'VWO'], comparison=True)

# Show the comparison
etf_comparison
```

Which returns the following:

|                                               | VOO          | QQQ          | ARKG       | VUG          | SCHA        | VWO                       |
|:----------------------------------------------|:-------------|:-------------|:-----------|:-------------|:------------|:--------------------------|
| ('sector_holdings', 'realestate')             | 2.75%        | 0.29%        | 0%         | 2.55%        | 7.16%       | 2.95%                     |
| ('sector_holdings', 'consumer_cyclical')      | 10.13%       | 14.2%        | 0%         | 18.09%       | 12.75%      | 12.92%                    |
| ('sector_holdings', 'basic_materials')        | 2.4%         | 0%           | 0%         | 2.01%        | 4.35%       | 9.34%                     |
| ('sector_holdings', 'consumer_defensive')     | 7.38%        | 6.67%        | 0%         | 2.92%        | 4.12%       | 6.23%                     |
| ('sector_holdings', 'technology')             | 23.65%       | 47.62%       | 3.41%      | 41.19%       | 14.18%      | 15.49%                    |
| ('sector_holdings', 'communication_services') | 7.43%        | 15.99%       | 0%         | 11.96%       | 2.74%       | 8.72%                     |
| ('sector_holdings', 'financial_services')     | 13.7%        | 0.74%        | 0%         | 6.95%        | 15.68%      | 20.63%                    |
| ('sector_holdings', 'utilities')              | 2.43%        | 0.88%        | 0%         | 0%           | 1.83%       | 3.78%                     |
| ('sector_holdings', 'industrials')            | 8.82%        | 4.75%        | 0%         | 4.64%        | 16.16%      | 7.26%                     |
| ('sector_holdings', 'energy')                 | 5.11%        | 0.49%        | 0%         | 1.35%        | 5.85%       | 5.36%                     |
| ('sector_holdings', 'healthcare')             | 15.27%       | 7.54%        | 96.58%     | 8.13%        | 13.98%      | 4.6%                      |
| ('annual_returns', '2022')                    | -18.15%      | -32.49%      | -53.94%    | -33.13%      | -19.8%      | -17.72%                   |
| ('annual_returns', '2021')                    | 28.66%       | 27.24%       | -33.89%    | 27.26%       | 16.35%      | 0.96%                     |
| ('annual_returns', '2020')                    | 18.35%       | 48.6%        | 180.51%    | 40.16%       | 19.35%      | 15.32%                    |
| ('annual_returns', '2019')                    | 31.46%       | 39.12%       | 43.75%     | 37.26%       | 26.54%      | 20.4%                     |
| ('annual_returns', '2018')                    | -4.42%       | -0.14%       | 0.59%      | -3.32%       | -11.75%     | -14.57%                   |
| ('annual_returns', '2017')                    | 21.78%       | 32.7%        | 45.41%     | 27.8%        | 15.04%      | 31.38%                    |
| ('key_characteristics', 'fundInceptionDate')  | 2010-09-07   | 1999-03-10   | 2014-10-31 | 2004-01-26   | 2009-11-03  | 2005-03-04                |
| ('key_characteristics', 'category')           | Large Blend  | Large Growth | Health     | Large Growth | Small Blend | Diversified Emerging Mkts |
| ('key_characteristics', 'totalAssets')        | 744769716224 | 145931501568 | 1899108352 | 132303921152 | 13327223808 | 93044613120               |
| ('key_characteristics', 'currency')           | USD          | USD          | USD        | USD          | USD         | USD                       |
| ('key_characteristics', 'navPrice')           | 366.24       | 281.03       | 32.79      | 225.08       | 43.42       | 41.9                      |
| ('key_characteristics', 'previousClose')      | 365.67       | 281.54       | 33.43      | 225.66       | 43.41       | 41.92                     |
| ('risk_data_3y', 'year')                      | 3y           | 3y           | 3y         | 3y           | 3y          | 3y                        |
| ('risk_data_3y', 'alpha')                     | -0.04        | 0.76         | -4.62      | -1.7         | -3.72       | -1.28                     |
| ('risk_data_3y', 'beta')                      | 1            | 1.08         | 1.32       | 1.11         | 1.15        | 0.9                       |
| ('risk_data_3y', 'meanAnnualReturn')          | 0.8          | 0.92         | 0.25       | 0.74         | 0.6         | 0.05                      |
| ('risk_data_3y', 'rSquared')                  | 100          | 87.57        | 41.82      | 90.79        | 82.66       | 76.68                     |
| ('risk_data_3y', 'stdDev')                    | 21.17        | 24.49        | 41.28      | 24.6         | 26.8        | 20.04                     |
| ('risk_data_3y', 'sharpeRatio')               | -0.55        | -0.7         | 0.27       | -0.72        | 1.58        | 2                         |
| ('risk_data_3y', 'treynorRatio')              | 6.76         | 7.04         | -4.51      | 4.7          | 2.44        | -2.53                     |
| ('risk_data_5y', 'year')                      | 5y           | 5y           | 5y         | 5y           | 5y          | 5y                        |
| ('risk_data_5y', 'alpha')                     | -0.04        | 2.15         | 2.44       | -0.2         | -5.06       | -0.93                     |
| ('risk_data_5y', 'beta')                      | 1            | 1.1          | 1.5        | 1.1          | 1.16        | 0.94                      |
| ('risk_data_5y', 'meanAnnualReturn')          | 0.89         | 1.16         | 0.98       | 0.96         | 0.6         | 0.11                      |
| ('risk_data_5y', 'rSquared')                  | 100          | 88.49        | 45.66      | 91.7         | 83.48       | 78.97                     |
| ('risk_data_5y', 'stdDev')                    | 18.69        | 21.85        | 39.32      | 21.4         | 23.73       | 18.27                     |
| ('risk_data_5y', 'sharpeRatio')               | -0.19        | -0.44        | -0.12      | -0.3         | 1.77        | 1.58                      |
| ('risk_data_5y', 'treynorRatio')              | 8.04         | 9.81         | 1.98       | 7.55         | 2.69        | -1.74                     |
| ('risk_data_10y', 'year')                     | 10y          | 10y          | 10y        | 10y          | 10y         | 10y                       |
| ('risk_data_10y', 'alpha')                    | -0.04        | 2.46         | 0          | -0.51        | -4.08       | -1.79                     |
| ('risk_data_10y', 'beta')                     | 1            | 1.1          | 0          | 1.08         | 1.16        | 0.98                      |
| ('risk_data_10y', 'meanAnnualReturn')         | 1.08         | 1.39         | 0          | 1.13         | 0.9         | 0.24                      |
| ('risk_data_10y', 'rSquared')                 | 100          | 85.33        | 0          | 91.48        | 80.44       | 75.61                     |
| ('risk_data_10y', 'stdDev')                   | 14.78        | 17.56        | 0          | 16.76        | 19.06       | 16.38                     |
| ('risk_data_10y', 'sharpeRatio')              | 0.88         | 0.39         | 0          | 0.78         | 2.92        | 1.44                      |
| ('risk_data_10y', 'treynorRatio')             | 11.7         | 14.01        | 0          | 11.05        | 7.38        | 0.79                      |

### PR DESCRIPTION
**Pull request by jmaslek**
Fixes #4014 and addresses issues that are encountered when running `safety check -r poetry.lock` and `safety check -r requirements-full.txt`. When run, there are two issues, `py` which is an open issue with pylint or pytest it looks like, and `protobuf` which is related to the forecasting build that I am not touching:

TODO: Rebuild installers

```

-> Vulnerability found in protobuf version 3.20.1
   Vulnerability ID: 51167
   Affected spec: >=3.20.0rc0,<3.20.2
   ADVISORY: Protobuf 3.18.3, 3.19.5, 3.20.2 and 4.21.6 include a fix for CVE-2022-1941: A parsing vulnerability for the
   MessageSet type in the ProtocolBuffers versions prior to and including 3.16.1, 3.17.3, 3.18.2, 3.19.4, 3.20.1 and 3.21.5 for...
   CVE-2022-1941
   For more information, please visit https://pyup.io/v/51167/f17


-> Vulnerability found in py version 1.11.0
   Vulnerability ID: 51457
   Affected spec: <=1.11.0
   ADVISORY: Py throughout 1.11.0 allows remote attackers to conduct a ReDoS (Regular expression Denial of Service) attack via a
   Subversion repository with crafted info data, because the InfoSvnCommand argument is mishandled.https://github.com/pytest-...
   CVE-2022-42969
   For more information, please visit https://pyup.io/v/51457/f17
```

The initial safety check was:
```
-> Vulnerability found in certifi version 2022.9.24
   Vulnerability ID: 52365
   Affected spec: <2022.12.07
   ADVISORY: Certifi 2022.12.07 includes a fix for CVE-2022-23491: Certifi 2022.12.07 removes root certificates from "TrustCor"
   from the root store. These are in the process of being removed from Mozilla's trust store. TrustCor's root certificates are being...
   CVE-2022-23491
   For more information, please visit https://pyup.io/v/52365/f17


-> Vulnerability found in cryptography version 38.0.2
   Vulnerability ID: 52173
   Affected spec: >=37.0.0,<38.0.3
   ADVISORY: Cryptography versions from 37.0.0 and before 38.0.2 include a statically linked copy of OpenSSL that has known
   vulnerabilities.https://github.com/pyca/cryptography/security/advisories/GHSA-39hc-v87j-747x
   CVE-2022-3786
   For more information, please visit https://pyup.io/v/52173/f17


-> Vulnerability found in cryptography version 38.0.2
   Vulnerability ID: 52174
   Affected spec: >=37.0.0,<38.0.3
   ADVISORY: Cryptography versions from 37.0.0 and before 38.0.2 include a statically linked copy of OpenSSL that has known
   vulnerabilities.https://github.com/pyca/cryptography/security/advisories/GHSA-39hc-v87j-747x
   CVE-2022-3602
   For more information, please visit https://pyup.io/v/52174/f17


-> Vulnerability found in future version 0.18.2
   Vulnerability ID: 52510
   Affected spec: <=0.18.2
   ADVISORY: Future 0.18.2 and earlier allows remote attackers to cause a denial of service via crafted Set-Cookie header from
   malicious web server.https://github.com/PythonCharmers/python-...
   CVE-2022-40899
   For more information, please visit https://pyup.io/v/52510/f17


-> Vulnerability found in gitpython version 3.1.29
   Vulnerability ID: 52322
   Affected spec: <3.1.30
   ADVISORY: Gitpython 3.1.30 includes a fix for CVE-2022-24439: Gitpython is vulnerable to Remote Code Execution (RCE) due to
   improper user input validation, which makes it possible to inject a maliciously crafted remote URL into the clone command....
   CVE-2022-24439
   For more information, please visit https://pyup.io/v/52322/f17


-> Vulnerability found in pillow version 9.2.0
   Vulnerability ID: 51886
   Affected spec: <9.3.0
   ADVISORY: Pillow before 9.3.0 allows denial of service via SAMPLESPERPIXEL.
   CVE-2022-45199
   For more information, please visit https://pyup.io/v/51886/f17


-> Vulnerability found in protobuf version 3.20.1
   Vulnerability ID: 51167
   Affected spec: >=3.20.0rc0,<3.20.2
   ADVISORY: Protobuf 3.18.3, 3.19.5, 3.20.2 and 4.21.6 include a fix for CVE-2022-1941: A parsing vulnerability for the
   MessageSet type in the ProtocolBuffers versions prior to and including 3.16.1, 3.17.3, 3.18.2, 3.19.4, 3.20.1 and 3.21.5 for...
   CVE-2022-1941
   For more information, please visit https://pyup.io/v/51167/f17


-> Vulnerability found in py version 1.11.0
   Vulnerability ID: 51457
   Affected spec: <=1.11.0
   ADVISORY: Py throughout 1.11.0 allows remote attackers to conduct a ReDoS (Regular expression Denial of Service) attack via a
   Subversion repository with crafted info data, because the InfoSvnCommand argument is mishandled.https://github.com/pytest-...
   CVE-2022-42969
   For more information, please visit https://pyup.io/v/51457/f17


-> Vulnerability found in pydantic version 1.8.2
   Vulnerability ID: 50916
   Affected spec: <1.10.2
   ADVISORY: Pydantic 1.10.2 prevents long strings as int inputs to fix
   CVE-2020-10735.https://github.com/pydantic/pydantic/commit/eccd85e4d012e70ffbd81f379179da900d4621c5
   CVE-2020-10735
   For more information, please visit https://pyup.io/v/50916/f17

 Scan was completed. 9 vulnerabilities were found. 
```

### CODE REVIEW

File: build/conda/conda-3-9-env-full.yaml [hunk 1]
```diff
@@ -6,4 +6,3 @@ dependencies:
   - pip
   - poetry=1.1.13
   - lightgbm=3.3.3
-  - charset-normalizer=2.1.1
```
Comment (jmaslek) [inline build/conda/conda-3-9-env-full.yaml:9]: Im not sure why this is in here.

Comment (piiq) [reply build/conda/conda-3-9-env-full.yaml:9]: there were weird packaging errors related to the versioning of this package. let's try removing the pin and see if that works out

Comment (jmaslek) [reply build/conda/conda-3-9-env-full.yaml:9]: So if you look at the versioning, I added this in the excel pr.  Was accidental and an issue with my rebase.

Hence why I removed it.

Comment (piiq) [reply build/conda/conda-3-9-env-full.yaml:9]: 🚀

**Review by luqmanbello — APPROVED** [review]

**Pull request merged**